core/field_vertical_separator.js

```diff
@@ -22,140 +22,133 @@
  * @extends {Blockly.Field}
  * @constructor
  */
-Blockly.FieldVerticalSeparator = function() {
-  this.sourceBlock_ = null;
-  this.width_ = 1;
-  this.height_ = Blockly.BlockSvg.ICON_SEPARATOR_HEIGHT;
-  this.size_ = new goog.math.Size(this.width_, this.height_);
-};
-goog.inherits(Blockly.FieldVerticalSeparator, Blockly.Field);
-
-/**
- * Construct a FieldVerticalSeparator from a JSON arg object.
- * @param {!Object} _element A JSON object with options (unused, but passed in
- *     by Field.fromJson).
- * @returns {!Blockly.FieldVerticalSeparator} The new field instance.
- * @package
- * @nocollapse
- */
-Blockly.FieldVerticalSeparator.fromJson = function(
-    /* eslint-disable no-unused-vars */ _element
-    /* eslint-enable no-unused-vars */) {
-  return new Blockly.FieldVerticalSeparator();
-};
-/**
- * Editable fields are saved by the XML renderer, non-editable fields are not.
- */
-Blockly.FieldVerticalSeparator.prototype.EDITABLE = false;
-
-/**
- * Install this field on a block.
- */
-Blockly.FieldVerticalSeparator.prototype.init = function() {
-  if (this.fieldGroup_) {
-    // Image has already been initialized once.
-    return;
-  }
-  // Build the DOM.
-  /** @type {SVGElement} */
-  this.fieldGroup_ = Blockly.utils.createSvgElement('g', {}, null);
-  if (!this.visible_) {
-    this.fieldGroup_.style.display = 'none';
+class FieldVerticalSeparator extends Blockly.Field {
+  constructor() {
+    super(Blockly.Field.SKIP_SETUP);
+    /**
+     * Editable fields are saved by the XML renderer, non-editable fields are not.
+     */
+    this.EDITABLE = false;
   }
-  /** @type {SVGElement} */
-  this.lineElement_ = Blockly.utils.createSvgElement('line',
-      {
-        'stroke': this.sourceBlock_.getColourSecondary(),
-        'stroke-linecap': 'round',
-        'x1': 0,
-        'y1': 0,
-        'x2': 0,
-        'y2': this.height_
-      }, this.fieldGroup_);
 
-  this.sourceBlock_.getSvgRoot().appendChild(this.fieldGroup_);
-};
+  /**
+   * Construct a FieldVerticalSeparator from a JSON arg object.
+   * @param {!Object} _element A JSON object with options (unused, but passed in
+   *     by Field.fromJson).
+   * @returns {!Blockly.FieldVerticalSeparator} The new field instance.
+   * @package
+   * @nocollapse
+   */
+  static fromJson = function(
+      /* eslint-disable no-unused-vars */ _element
+      /* eslint-enable no-unused-vars */) {
+    return new FieldVerticalSeparator();
+  }
 
-/**
- * Set the height of the line element, without adjusting the field's height.
- * This allows the line's height to be changed without causing it to be
- * centered with the new height (needed for correct rendering of hat blocks).
- * @param {number} newHeight the new height for the line.
- * @package
- */
-Blockly.FieldVerticalSeparator.prototype.setLineHeight = function(newHeight) {
-  this.lineElement_.setAttribute('y2', newHeight);
-};
+  /**
+   * Install this field on a block.
+   */
+  init() {
```

Comment:
You shouldn't need to override `init` instead you should be able to use `initView`. More docs: https://developers.google.com/blockly/guides/create-custom-blocks/fields/customizing-fields/creating#initializing

Reply:
Done